### PR TITLE
Factor regional office and end product methods into classes

### PR DESCRIPTION
**Pull request by shanear**
A couple of refactors to make a smoother ride for Intake:

- There was a bunch of dead code surrounding the RO page. We used that in the original rollout of Caseflow but it's pointless now.
- Don't pretend we load RegionalOffice info from VACOLS. We use hard coded values, refactor the code to make that clear.
- Move a bunch of logic that was buried in specific classes out into places where they make more sense and are easier to reuse, such as `RegionalOffice` and `EndProduct`

NOTE: There are more places we can use `RegionalOffice` to abstract some of the logic from other apps, but this PR is already sort of a doozy, so I'll follow up with that one

connects #3749

### CODE REVIEW

File: app/controllers/offices_controller.rb [hunk 1]
```diff
@@ -1,7 +0,0 @@
-class OfficesController < ApplicationController
```
Comment (shanear) [inline app/controllers/offices_controller.rb]: This controller was not in routes and no longer used

File: app/models/appeal.rb [hunk 2]
```diff
@@ -227,11 +227,6 @@ def attributes_for_hearing
     }
   end
 
-  def station_key
```
Comment (shanear) [inline app/models/appeal.rb]: moved to `regional_office.rb`

File: app/models/claim_establishment.rb [hunk 3]
```diff
@@ -25,17 +25,7 @@ def self.get_decision_type(appeal)
     DECSION_TYPES[appeal.decision_type]
   end
 
-  def ep_description
```
Comment (shanear) [inline app/models/claim_establishment.rb]: moved to `end_product.rb`

File: app/models/concerns/appeal_concern.rb [hunk 4]
```diff
@@ -1,13 +1,14 @@
 module AppealConcern
   extend ActiveSupport::Concern
 
+  delegate :station_key, to: :regional_office
+
   def regional_office
-    { key: regional_office_key }.merge(VACOLS::RegionalOffice::CITIES[regional_office_key] ||
-                                       VACOLS::RegionalOffice::SATELLITE_OFFICES[regional_office_key] || {})
```
Comment (shanear) [inline app/models/concerns/appeal_concern.rb]: this logic is now captured in `RegionalOffice.location_hash` and `RegionalOffice.to_h`

File: app/models/regional_office.rb [hunk 5]
```diff
@@ -1,39 +1,238 @@
 class RegionalOffice
-  include ActiveModel::Model
+  class NotFoundError < StandardError; end
 
-  attr_accessor :ro_id
-  attr_accessor :city, :state
-  attr_accessor :caseflow_enabled
```
Comment (shanear) [inline app/models/regional_office.rb]: unused

File: app/models/regional_office.rb [hunk 6]
```diff
@@ -1,39 +1,238 @@
 class RegionalOffice
-  include ActiveModel::Model
+  class NotFoundError < StandardError; end
 
-  attr_accessor :ro_id
-  attr_accessor :city, :state
```
Comment (shanear) [inline app/models/regional_office.rb]: these are now calculated dynamically

File: app/models/regional_office.rb [hunk 7]
```diff
@@ -1,39 +1,238 @@
...
-  attr_accessor :caseflow_enabled
+  # Maps CSS Station # to RO id
```
Comment (shanear) [inline app/models/regional_office.rb]: moved all the constants from `VACOLS::RegionalOffice` here

File: app/services/regional_office_repository.rb [hunk 8]
```diff
@@ -1,21 +0,0 @@
-class RegionalOfficeRepository
```
Comment (shanear) [inline app/services/regional_office_repository.rb]: We never loaded regional offices from VACOLS. This was unused

File: app/models/concerns/appeal_concern.rb [hunk 9]
```diff
@@ -1,13 +1,16 @@
 module AppealConcern
   extend ActiveSupport::Concern
 
+  delegate :station_key, to: :regional_office
```
Comment (amprokop) [inline app/models/concerns/appeal_concern.rb]: whoa, cool. didn't know about this method

File: app/models/concerns/appeal_concern.rb [hunk 10]
```diff
@@ -1,13 +1,16 @@
 module AppealConcern
```
Comment (amprokop) [inline app/models/concerns/appeal_concern.rb]: This is outside the scope of this PR probably, but I'm wondering your opinion — is this concern the optimal separation of logic? I'm not sure what it really encapsulates. I know @aroltsch's goal was to break up the appeal model, but is this something that should be rethought?

File: app/models/concerns/appeal_concern.rb [hunk 11]
```diff
@@ -1,13 +1,16 @@
...
   extend ActiveSupport::Concern
 
+  delegate :station_key, to: :regional_office
+
   def regional_office
-    { key: regional_office_key }.merge(VACOLS::RegionalOffice::CITIES[regional_office_key] ||
-                                       VACOLS::RegionalOffice::SATELLITE_OFFICES[regional_office_key] || {})
+    # TODO: We should probably use RegionalOffice.find! here since it
```
Comment (amprokop) [inline app/models/concerns/appeal_concern.rb]: Why can't we use `find!`now? Are we not sure if it's a risky change?

File: app/models/end_product.rb [hunk 12]
```diff
@@ -121,6 +137,10 @@ def active?
     !INACTIVE_STATUSES.include?(status_type_code)
   end
 
+  def regional_office
+    @regional_office ||= RegionalOffice.for_station(station_of_jurisdiction).first
```
Comment (amprokop) [inline app/models/end_product.rb]: hmm. is it appropriate to always return the first station in the list?

Comment (shanear) [reply app/models/end_product.rb]: Yeah, that's what dispatch was doing. See:

https://github.com/department-of-veterans-affairs/caseflow/pull/3821/files#diff-d38fdc00b17f61e0a6e1bd027f26224cL232

File: app/models/concerns/appeal_concern.rb (hunk 11 above)
Comment (shanear) [reply app/models/concerns/appeal_concern.rb]: it was breaking some unit tests, it may throw some errors in some edge cases i'm not sure about. I'll add it back in. #confidence

File: app/models/concerns/appeal_concern.rb (hunk 10 above)
Comment (shanear) [reply app/models/concerns/appeal_concern.rb]: I agree. The reason is because the concern is also on `Hearing`. It's kind of janky, but servicible lol, and prevents a bunch of duplication.

**Comment by shanear** [comment]
> Question — do you think we should move in the direction of packaging up all our VACOLS code into a gem? It’s something we’ve explored in the past. If we want to experiment with creating Intake and potentially Work Queue in their own services, it might be a necessary first step. Would love to bring up that topic, this looks good to me though.

Yeah, I think me, you and sunil will be talking about how best to do this in the coming weeks

File: app/services/regional_office_repository.rb (hunk 8 above)
Comment (shanear) [reply app/services/regional_office_repository.rb]: We don't currently, but it also never changes (since I've been here it hasn't)
